### DISCUSSION TITLE
How to configure https_proxy to multiple values

### DISCUSSION THREAD

**Discussion by ZhangQing95** [moby/moby#45580] [Q&A]
How to configure https_proxy to multiple values

**Comment by thaJeztah** [comment]
It's not possible to specify multiple proxies in that configuration.

The `HTTP_PROXY`, `http_proxy` (and other variants, such as `HTTPS_PROXY`, `FTP_PROXY`) environment variables don't have an  "official" specification, but are mostly a de-facto standard, and I don't think there's any software that would allow it to have multiple proxies. Docker itself for sure only allows for a single proxy to be configured (Docker is written in Go, and uses the formats described in https://pkg.go.dev/golang.org/x/net@v0.10.0/http/httpproxy#Config).

**Reply by ZhangQing95** [reply]
Thank you for your answer. 
It's like this, we would like to add HTTPS_ PROXY is configured as our own service. So, we want to ensure that when there is a problem with our own proxy service, Docker's image pulling can still be performed normally, similar to containerd being able to configure multiple endpoints for backup. 
I don't know if Docker has this kind of guarantee

**Reply by thaJeztah** [reply]
Do you mean the "mirrors" configuration for registries in containerd? https://github.com/containerd/containerd/blob/878132923db35f31c8c7b324d9cd1e5f59046064/docs/hosts.md#setup-a-local-mirror-for-docker
